cli/internal/cache/cache_fs.go

```diff
@@ -89,7 +89,7 @@ func (f *fsCache) Put(target, hash string, duration int, files []string) error {
 						return fmt.Errorf("error ensuring directory file from cache: %w", err)
 					}
 
-					if err := fs.CopyOrLinkFile(file, filepath.Join(f.cacheDirectory, hash, file), fromInfo.Mode(), fs.DirPermissions, false, false); err != nil {
+					if err := fs.CopyOrLinkFile(&fs.LstatCachedFile{Path: fs.AbsolutePath(file), Info: fromInfo}, filepath.Join(f.cacheDirectory, hash, file), false, false); err != nil {
```

Comment:
Let's create the `LstatCachedFile` first, with just the path (at the top of this for loop), and then call `GetInfo()` on it to populate it. Then at this site we can just past that reference forward.

(That way we have a common interface for as much of this as possible.)

Reply:
Good call, We actually only need a `GetType` since we only needed `GetInfo` for access to `isDir()` which I can now just use the type for (`fromType != os.ModeDir).

Your statement still stands as we can move this up to be consistent and populate the interface.